presto-main/src/test/java/com/facebook/presto/operator/TestPartitionedOutputOperator.java

```diff
@@ -0,0 +1,311 @@
+        assertEquals(outputPositionCount, 51_200_000);
+        assertTrue(outputSize > createPageWithDictionary().getSizeInBytes() * NUM_OF_ITERATIONS);
+    }
+
+    @Test
+    public void testOutputForPageWithRunLengthAndReplication()
+    {
+        PartitionedOutputOperator partitionedOutputOperator = createPartitionedOutputOperator(true);
+        for (int i = 0; i < NUM_OF_ITERATIONS; i++) {
+            partitionedOutputOperator.addInput(createPageWithRunLengthAndNullBlock());
+        }
+        partitionedOutputOperator.finish();
+
+        OperatorContext operatorContext = partitionedOutputOperator.getOperatorContext();
+        long outputSize = operatorContext.getOutputDataSize().getTotalCount();
+        long outputPositionCount = operatorContext.getOutputPositions().getTotalCount();
+        assertEquals(outputSize, 461_260_800);
+        assertEquals(outputPositionCount, 51_200_000);
+        assertTrue(outputSize > createPageWithRunLength().getSizeInBytes() * NUM_OF_ITERATIONS);
+    }
+
+    private PartitionedOutputOperator createPartitionedOutputOperator(boolean shouldReplicate)
+    {
+        PartitionFunction partitionFunction = new LocalPartitionGenerator(new InterpretedHashGenerator(ImmutableList.of(BIGINT), new int[] {0}), PARTITION_COUNT);
+        PagesSerdeFactory serdeFactory = new PagesSerdeFactory(new BlockEncodingManager(new TypeRegistry()), false);
+        PartitionedOutputBuffer buffer = createPartitionedOutputBuffer();
+        PartitionedOutputOperator.PartitionedOutputFactory operatorFactory;
+        if (shouldReplicate) {
+            operatorFactory = new PartitionedOutputOperator.PartitionedOutputFactory(
+                    partitionFunction,
+                    ImmutableList.of(0),
+                    ImmutableList.of(Optional.empty()),
+                    true,
+                    OptionalInt.of(0),
+                    buffer,
+                    PARTITION_MAX_MEMORY);
+            return (PartitionedOutputOperator) operatorFactory
+                    .createOutputOperator(0, new PlanNodeId("plan-node-0"), REPLICATION_TYPES, Function.identity(), serdeFactory)
+                    .createOperator(createDriverContext());
+        }
+        else {
+            operatorFactory = new PartitionedOutputOperator.PartitionedOutputFactory(
+                    partitionFunction,
+                    ImmutableList.of(0),
+                    ImmutableList.of(Optional.empty(), Optional.empty()),
+                    false,
+                    OptionalInt.empty(),
+                    buffer,
+                    PARTITION_MAX_MEMORY);
+            return (PartitionedOutputOperator) operatorFactory
+                    .createOutputOperator(0, new PlanNodeId("plan-node-0"), TYPES, Function.identity(), serdeFactory)
+                    .createOperator(createDriverContext());
+        }
+    }
+
+    private PartitionedOutputBuffer createPartitionedOutputBuffer()
```

Comment:
make this method static

Reply:
Wondering what is the rationale to mark it as `static` ? . If it is though of as a utility method then probably we need to move this to a better home in one of the helper classes like the  `TestingBuilders`, Ying is working on.

Reply:
Methods that don't need access to instance variables should be marked `static`. This way there is only one copy of the method and individual object instances don't need to have separate copies of it.